connectors/src/connectors/salesforce/index.ts

```diff
@@ -41,15 +66,48 @@ export class SalesforceConnectorManager extends BaseConnectorManager<null> {
   }: {
     connectionId?: string | null;
   }): Promise<Result<string, ConnectorManagerError<UpdateConnectorErrorCode>>> {
-    // TODO(salesforce): implement this
+    // Get connector.
+    const c = await ConnectorResource.fetchById(this.connectorId);
+    if (!c) {
+      logger.error({ connectorId: this.connectorId }, "Connector not found");
+      throw new Error(`Connector ${this.connectorId} not found`);
+    }
 
-    void connectionId;
+    // If no connection ID is provided, we return the current connector ID.
+    if (!connectionId) {
+      return new Ok(c.id.toString());
+    }
 
-    const connector = await ConnectorResource.fetchById(this.connectorId);
-    if (!connector) {
-      throw new Error(`Connector ${this.connectorId} not found`);
+    // Get credentials.
+    const credentialsRes = await getSalesforceCredentials(connectionId);
+    if (credentialsRes.isErr()) {
+      throw credentialsRes.error;
     }
-    return new Ok(connector.id.toString());
+
+    // Test connection.
+    const connectionRes = await testSalesforceConnection(credentialsRes.value);
+    if (connectionRes.isErr()) {
+      return new Err(
+        new ConnectorManagerError(
+          "INVALID_CONFIGURATION",
```

Comment:
It's not really an invalid config since there's no config for this one, not sure which one is best

Reply:
I think it's fine 👍